src/modules/auth/credentials/BulkAssignmentContainer.js

```diff
@@ -0,0 +1,485 @@
+
+	const resource = "authz:superuser";
+	const resources = useSelector(state => state.auth?.resources);
+
+	// headers for Credentails List
+	const headers = [
+		{
+			name: '',
+			customComponent: {
+				generate: (credential) => (
+					<div>
+						<ButtonWithAuthz
+							title={t(`BulkAssignmentContainer|${credential.assigned ? "Credential already assigned" : "Add credential"}`)}
+							id={credential._id}
+							size="sm"
+							color="primary"
+							outline
+							onClick={() => saveToSelectedCredentials(credential)}
+							resource={resource}
+							resources={resources}
+							disabled={credential.assigned}
+						>
+							<i className="cil-plus"></i>
+						</ButtonWithAuthz>
+					</div>
+				)
+			}
+		},
+		{
+			name: '',
+			customComponent: {
+				generate: (obj) => (
+					<div className="no-wrap-40ch">
+						{(obj.suspended === true) ?
+							<span className="cil-user-unfollow text-muted mr-1" title={(obj.registered === false) ? t("BulkAssignmentContainer|Credentials invited") : t("BulkAssignmentContainer|Credentials suspended")}/>
+							: <span className="cil-user mr-1" />}
+						<Link
+							className={obj.suspended ? 'credentials-suspended' : null}
+							onClick={() => confirmForm(`/auth/credentials/${obj._id}`)}
+							>
+							{/* TODO: substitute with Credentials component, when it's ready */}
+							{obj.username ?? obj._id}
+						</Link>
+					</div>
+				)
+			},
+		}
+	];
+
+	// headers for Tenants List
+	const tenantHeaders = [
+		{
+			name: '',
+			customComponent: {
+				generate: (tenant) => (
+					<div>
+						<ButtonWithAuthz
+							title={t(`BulkAssignmentContainer|${tenant.assigned ? "Tenant already assigned" : "Add tenant"}`)}
+							id={tenant._id}
+							size="sm"
+							color="primary"
+							outline
+							onClick={() => saveToSelectedTenants(tenant)}
+							resource={resource}
+							resources={resources}
+							disabled={tenant.assigned}
+						>
+							<i className="cil-plus"></i>
+						</ButtonWithAuthz>
+					</div>
+				)
+			}
+		},
+		{
+			name: '',
+			customComponent: {
+				generate: (tenant) => (
+					<Link onClick={() => confirmForm(`/auth/tenant/${tenant._id}`)}>
+						{tenant._id}
+					</Link>
+				)
+			}
+		}
+	];
+
+	// UseEffect to fetch data for Credentials List based on changes in page/credentialsFilter/limit
+	useEffect(() => {
+		if (limit > 0) {
+			retrieveData();
+		}
+	}, [page, credentialsFilter, limit]);
+
+	// UseEffect to fetch data for Tenants List based on changes in page/tenantsFilter/limit
+	useEffect(() => {
+		if (limit > 0) {
+			retrieveTenants();
+		}
+	}, [tenantsPage, tenantsFilter, tenantsLimit]);
+
+	/* Actual modified data (this data set was compared compared with `selectedCredentials` and `assigned: true` was added to matching credentials. Assigned:true property
+	disables the "+"" button) to be displayed in Credentials list data table */
+	const datatableCredentialsData = useMemo(() => {
```

Comment:
@petrKavulok Since you have duplicit code in `datatableCredentialsData` and `datatableTenantsData`, I propose to unify the guts of it:

create a new method, which will return desired result:

```
const matchAssigned = (data, selectedEvent) => {
	let tableData = [];
	if (data) {
		data.map((dataObj) => {
			let matchedObj = selectedEvent.find(obj => obj._id === dataObj._id);
			if (matchedObj) {
				matchedObj['assigned'] = true;
				tableData.push(matchedObj);
			} else {
				dataObj['assigned'] = false;
				tableData.push(dataObj);
			}
		})
	}
	return tableData;
}
```

and use it within the memoised function

```
const datatableCredentialsData = useMemo(() => {
return matchAssigned(data, selectedCredentials);
}, [data, selectedCredentials]);
```

and repeat for `datatableTenantsData`

Always think about your code - dont repeat yourself or simplify what can be simplified or done more performant :)